src/TypeInference/TypeUtils.ml

```diff
@@ -48,7 +48,8 @@ let open_scheme ~pos (env : Env.t) (sch : T.scheme) =
         let (env, x) =
           match name with
           | T.NLabel -> Env.add_the_label_sch env sch
-          | T.NVar x -> Env.add_poly_var env x sch
+          | T.NVar x  -> Env.add_poly_var env x sch
+          | T.NOptionalVar x -> Env.add_mono_var env x sch.sch_body
```

Comment:
`add_mono_var env x sch.sch_body` and `add_poly_var env x sch` are equivalent for monomorphic types (for polymorphic types the former doesn't have any sense). Maybe these two clauses could be merged?